### ISSUE TITLE
ZeroDivisionError: float division by zero

### ISSUE THREAD

**Issue by heya5** [nalepae/pandarallel#184]
## General

- **Operating System**: windows 11
- **Python version**: 3.8.3
- **Pandas version**:  1.4.2
- **Pandarallel version**:  1.6.1

## Acknowledgement

- [ ] My issue is **NOT** present when using `pandas` without alone (without `pandarallel`)
- [ ] If I am on **Windows**, I read the [Troubleshooting page](https://nalepae.github.io/pandarallel/troubleshooting/)
      before writing a new bug report

## Bug description
WARNING: You are on Windows. If you detect any issue with pandarallel, be sure you checked out the Troubleshooting page:
   0.99%                                          |       46 /     4628 |
   0.00%                                          |        0 /     4627 |                                                                                                                               multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\multiprocessing\pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandarallel\core.py", line 158, in __call__
    results = self.work_function(
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandarallel\data_types\series.py", line 26, in work
    return data.apply(
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandas\core\series.py", line 4433, in apply
    return SeriesApply(self, func, convert_dtype, args, kwargs).apply()
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandas\core\apply.py", line 1082, in apply
    return self.apply_standard()
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandas\core\apply.py", line 1137, in apply_standard
    mapped = lib.map_infer(
  File "pandas\_libs\lib.pyx", line 2870, in pandas._libs.lib.map_infer
  File "D:\Users\yh110\Anaconda3\envs\22_esci\lib\site-packages\pandarallel\progress_bars.py", line 206, in closure
    state.next_put_iteration += max(int((delta_i / delta_t) * 0.25), 1)
ZeroDivisionError: float division by zero
### Observed behavior

_Write here the observed behavior_

### Expected behavior

_Write here the expected behavior_

## Minimal but working code sample to ease bug fix for `pandarallel` team

_Write here the minimal code sample to ease bug fix for `pandarallel` team_

**Comment by felixsc1** [comment]
same problem here, one of the progress bars turns red at some percentage point (different every time) and stops progressing, then after the others complete, I get the division by zero error.  I don't have a code example right now...

**Comment by till-m** [comment]
Hi @heya5 @felixsc1,

Unfortunately I don't have access to a windows machine. Is there any chance one of you would be able to investigate and/or fix this bug?

**Comment by dacianpeng** [comment]
same problem here, and `progress_bar=False` do not fix this too

**Comment by versatile712** [comment]
I also run into the problem. Sometimes rerunning the codes helps sovle the it, but it wastes a lot of time. Hope it can be fixed.

**Comment by theGreen-Coder** [comment]
Same problem here! And I'm using macOS.

**Comment by till-m** [comment]
Hi @theGreen-Coder 

If you can provide a snippet that reproduces the problem I can try to fix the problem, I have access to an M1 Mac.

**Comment by theGreen-Coder** [comment]
@till-m Great!

I'll try to explain as best as I can. Here's the link to the exact code on Google Collab: https://colab.research.google.com/drive/1FUZXHNCXtqWlip0jQN1Z9QBJ7JBO5qIR?usp=sharing

When I try to run a wald test:
```
test = de.test.wald(
    data=data,
    formula_loc="~ 1 + condition",
    factor_loc_totest="condition",
)

# Error Terminates the program here on my Mac

summaryTest = test.summary().iloc[:10,:]
```

It works on Google Collab, but not on my MacOS. When I run it on Google Collab, it also gives me an error (below), but that doesn't stop the execution of my code. And therefore, `test` is defined, as well as `summaryTest`.
However, when I run this exact same code on my Mac machine, this happens (below). An exception is thrown and the whole program stops (leaving `test` undefined, thus leaving also `summaryTest` undefined).

### Error on Google Collab (last line)

![Screenshot 2022-11-11 at 00 28 23](https://user-images.githubusercontent.com/109465400/201234137-06acef1a-b564-426f-959e-ec000a7e30fd.png)

### Error on my Mac
<img width="1059" alt="Screenshot 2022-11-11 at 00 32 06" src="https://user-images.githubusercontent.com/109465400/201234523-1490fbb2-b3d4-4fb4-9f88-9f92594a73f5.png">
<img width="592" alt="Screenshot 2022-11-11 at 00 32 50" src="https://user-images.githubusercontent.com/109465400/201234622-6166a171-fd3d-4513-be01-0bc1b4fff9bd.png">

The only way this works on my Mac is if I ran this line of code before the wald test:
`sc.pp.highly_variable_genes(data, n_top_genes=255, subset = True, flavor = "seurat_v3", batch_key="sample")`

But there's no point in doing that, because that only tests 255 genes out of the almost 36000 genes on my dataset. Of course, I had no intention to test all the 36000 genes, but I would probably like to test more than 255 genes.

Let me know if you need more information or access to my dataset to run this on your computer. I have only been using diffxpy for a couple of days, so I apologise in advance if I made some sort of rookie mistake.

Thank you :)

**Comment by theGreen-Coder** [comment]
Oops. Yeah, sorry I thought this was on `diffxpy`. My bad.

**Comment by till-m** [comment]
should be fixed with #212. I will close this issue, if the problem persists please let me know and I will reopen.
